lib/Haskell/Extra/Dec.agda

```diff
@@ -0,0 +1,27 @@
+module Haskell.Extra.Dec where
+
+open import Haskell.Prelude hiding (Reflects)
```

Comment:
Do you think you could remove the old definition of `Reflects`? and maybe just define patterns synonyms for `ofY` and `ofN` if you don't want to touch `Haskell.Law` too much.

On a side note, I don't think anything from `Haskell.Law` should be exported in `Haskell.Prelude`.